File: 7.py
'''So much for one liners...

I had to make an incrementer(x=0) so I could numbers.pop(x).
Numbers.remove(number) was WAY less efficient than numbers.pop(x).
'''

def seiveOfEratosthenes(primeRange):
  numbers = [number for number in range(2,primeRange)]
  primes = []
  while numbers != []:
    prime = numbers.pop(0)
    primes.append(prime)
    x = 0
    for number in numbers:
      if number % prime == 0:
        numbers.pop(x)
      x+=1
  return primes

print(seiveOfEratosthenes(120000)[10001-1])

File: 7.hs
module Main where

seiveOfEratosthenes :: Int -> [Int]
seiveOfEratosthenes range = getPrimes [2..range]
  where getPrimes (number:numbers) = number : (getPrimes $ filter (\n -> n `mod` number /= 0) numbers)
        getPrimes [] = []

main = putStrLn $ show $ (seiveOfEratosthenes 120000) !! 10000
